hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/container/IncrementalContainerReportHandler.java

```diff
@@ -66,16 +67,25 @@ public void onMessage(final IncrementalContainerReportFromDatanode report,
         nodeManager.addContainer(dd, id);
         processContainerReplica(dd, replicaProto);
       } catch (ContainerNotFoundException e) {
+        success = false;
         LOG.warn("Container {} not found!", replicaProto.getContainerID());
       } catch (NodeNotFoundException ex) {
+        success = false;
         LOG.error("Received ICR from unknown datanode {} {}",
             report.getDatanodeDetails(), ex);
       } catch (IOException e) {
+        success = false;
         LOG.error("Exception while processing ICR for container {}",
             replicaProto.getContainerID());
       }
     }
 
+    if (success) {
+      getContainerManager().notifyContainerReportProcessing(false, true);
+    } else {
```

Comment:
Any reason we are not doing _getContainerManager().notifyContainerReportProcessing(false, success)_ ?

Reply:
We can do that, just due to copy/paste from ICR code it is like this. (End result is same)
Let me know if you want to change it?

Reply:
IMO, we should change it to 1 line instead of if-else.

Reply:
As discussed offline, will change this in later jira. For now we shall go with the current way.